### PR TITLE
Accumulated Changes from UKAEA Pilots

### PR DESCRIPTION
**Pull request by bielsnohr**
Overtime, we have caught a number of corrections and improvements while running the course at UKAEA. I think most of these are general enough that they merit making their way into this incubator version of the course.

This PR also adds the first version of the slides I have created. I am currently in the process of updating these slides for the 5-section version of the course, hence why I have made this a Work In Progress PR. However, I think it could be merged in its current state depending on what others think.

Alternatively, I could extract only the commits related to content modification and leave out the slides.

### CODE REVIEW

**Comment by anenadic** [comment]
Slides are a great addition - thank you @bielsnohr!

File: slides/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Note for using slides
+
+Some quick information about how to use the slides when delivering the course.
+
+The slides themselves are written in Jupyter notebooks which has the built-in capability to produce slides. At the moment, the final HTML is not stored, so you need to have Jupyter installed. A virtual environment with a subsequent `pip` command is recommended:
```
Comment (anenadic) [inline slides/README.md:5]:
```suggestion
The slides themselves are written in Jupyter notebooks, which have the built-in capability to produce slides. At the moment, the final HTML version of slides is not included, so you need to have Jupyter installed in order to render them. A virtual environment with a subsequent `pip` command is recommended to "run" the slides:
```

File: _episodes/24-diagnosing-issues-improving-robustness.md [hunk 2]
```diff
@@ -425,9 +425,9 @@ def patient_normalise(data):
 
     Negative values are rounded to 0.
     """
-    max = np.nanmax(data, axis=1)
+    maxima = np.nanmax(data, axis=1)
```
Comment (anenadic) [inline _episodes/24-diagnosing-issues-improving-robustness.md:428]: I think we do want to call this variable `max` as we want the linter to thrown the error about "Redefining built-in 'max'" later on.

**Review by anenadic — APPROVED** [review]
I am approving this PR even with the missing slide (I've left some comments for you to consider) - many thanks for all the improvements @bielsnohr. I'd just open another issue to remind us about adding the missing slide.

**Comment by anenadic** [comment]
@bielsnohr - noticed many new figures in this PR but cannot seem them included in the material?

File: _episodes/24-diagnosing-issues-improving-robustness.md (hunk 2 above)
Comment (bielsnohr) [reply _episodes/24-diagnosing-issues-improving-robustness.md:428]: Yes, you are correct. I'll remove this change.

Comment (bielsnohr) [reply _episodes/24-diagnosing-issues-improving-robustness.md:428]:
```suggestion
    max = np.nanmax(data, axis=1)
```

**Comment by bielsnohr** [comment]
> I am approving this PR even with the missing slide (I've left some comments for you to consider) - many thanks for all the improvements @bielsnohr. I'd just open another issue to remind us about adding the missing slide.

The slides for the next section are coming very soon. I need to create them for the current run of the course I am doing at UKAEA, so there is no way they won't exist by the end of this month :sweat_smile: Regardless, I will add an issue to track that explicitly.

> @bielsnohr - noticed many new figures in this PR but cannot seem them included in the material?

I think most of those will be figures included in the slides, but I will double check that none are left over from any experimentation. Would you prefer the figures for the slides to be in their own separate folder?

File: _episodes/24-diagnosing-issues-improving-robustness.md [hunk 3]
```diff
@@ -425,9 +425,9 @@ def patient_normalise(data):
...
     with np.errstate(invalid='ignore', divide='ignore'):
-        normalised = data / max[:, np.newaxis]
+        normalised = data / maxima[:, np.newaxis]
```
Comment (anenadic) [inline _episodes/24-diagnosing-issues-improving-robustness.md:430]: normalised = data / max[:, np.newaxis]

**Review by anenadic — APPROVED** [review]
One [small final fix](https://github.com/carpentries-incubator/python-intermediate-development/pull/252#discussion_r1389432042) and think this is ready - happy merging @bielsnohr!

File: _episodes/24-diagnosing-issues-improving-robustness.md (hunk 3 above)
Comment (bielsnohr) [reply _episodes/24-diagnosing-issues-improving-robustness.md]: Aha, yes this is what happens when I only look at the immediate vicinity on GitHub. Much easier to spot from my editor! Thanks for catching this.

**Pull request merged**